### DISCUSSION TITLE
Multi-target advanced door

### DISCUSSION THREAD

**Discussion by yobanyvrod** [CreativeMD/LittleTiles#962] [Ideas]
### An Advanced Door that could move, but to multiple targets, based on inputs.

An option could be added to the Advanced Door to add more movement graphs.
These graphs would act like the existing one but in the signals tab there could be multiple outputs for the movement of the door.
I'm not fully sure how to put this into text so I hope this picture helps.
<details><summary>Image</summary>
<p>

![image](https://github.com/CreativeMD/LittleTiles/assets/66252573/1c1a87b5-6835-4c7b-b748-25d1fb3a9562)

</p>
</details> 
Multiple outputs for the state of the door could be present that represents each of the movement graphs that could be triggered separately based on which location is best fit for the occasion.
e.g.,
If I had a TV I could make door activators on the left, right and upper edge of the frame to angle the TV. If I wanted the TV to be angled either left or right, I would be able to press the left and right activators respectively, and upper edge if straight. 
I would then link the different activators to their respective graphs as shown in the aforementioned picture.
I could link one activator to State Nr.1 which could rotate it one way while another activator connected to State Nr.2 would rotate it the other way, and so on...

The graphs could function in a way where they override each others' actions so that the TV would rotate from the current position to the target position by either moving to the default position first and then moving to the target position, OR moving straight to the target position OR just executing a move event from the current position (more info below)
<details><summary>Details</summary>
<p>

If the TV is rotated 30 degrees left, executing a move event that rotates the TV to the right 80 degrees would result in a final angle of 50 degrees to the right, and not 80 degrees from start.

Therefore, each graph could be tailored not as a target but simply a move function from the current position, where every graph would be a movement from theoretical target position to another, although it's obvious that this will get exponentially more complex the more target positions you have e.g., 16 movements from each 4 target positions to the next -4 for duplicates so 12 movements for 4 positions. _I think_

</p>
</details> 

<details><summary>Waffle</summary>
<p>

I'm not sure if I wrote this whole thing right or not because I've stayed up all night as I'm trying to fix my sleep schedule and I am delirious.😂

</p>
</details> 
As always, Thank you in advance for reading and considering my idea proposal.

**Comment by CreativeMD** [comment]
You could achieve such behavior with doors in doors. You could create 3 doors, each one of them moves in a certain direction. There are plenty of options to make this work out. Not really a fan of multiple outputs in that sense. There are plans to introduce new structures which allow for other things to be done. Problem is time at the moment. So I'm not sure when I will get to it.
